src/probe.cpp

```diff
@@ -27,7 +27,13 @@
 #include "src/common/exception.h"
 #include "src/probe.h"
 
-Probe::Probe() {}
+Probe::Probe() {
+	_pathPrefixes[kResultAlanWake] = "";
+	_pathPrefixes[kResultAlanWakeRemastered] = "d:/data/";
+	_pathPrefixes[kResultAmericanNightmare] = "d:/data/";
+	_pathPrefixes[kResultQuantumBreak] = "d:/data/";
+	_pathPrefixes[kResultControl] = "data/";
```

Comment:
@Nostritius I initially started look at using an array, but the reason I used a map is because of the way that ProbeResult is currently defined. Each game represents a bit in a bitmask which is used when processing the Probe result. That makes it a bit harder to convert to an array index in a clean way.
I did not want to change ProbeResult representation in this PR because it will also impact other parts.
If you agree I could make it a const map and take it out of the constructor? Unless there are other suggestions.

Reply:
> Each game represents a bit in a bitmask which is used when processing the Probe result. That makes it a bit harder to convert to an array index in a clean way.

Well, since the project started using C++20 features about a month ago, we can actually try and make an enum that doesn't use power-of-two values while not changing that much:

`src/probe.h:26`
```cpp
enum ProbeResult {
	kResultUnknown = 0,
	kResultAlanWake = 1,
	kResultAmericanNightmare = 2,
	kResultQuantumBreak = 3,
	kResultAlanWakeRemastered = 4,
	kResultControl = 5,
};
```

`src/probe.cpp:21`
```cpp
// Bit manipulation std lib from C++20, included for std::countr_zero
#include <bit>
#include <bitset>
```

`src/probe.cpp:38`
```cpp
ProbeResult Probe::performProbe() {
	spdlog::info("Probing archives...");
	std::bitset<8> allChecks;

	allChecks.set(checkAlanWake());
	allChecks.set(checkAmericanNightmare());
	allChecks.set(checkQuantumBreak());
	allChecks.set(checkAlanWakeRemastered());
	allChecks.set(checkControl());

	if (allChecks.none())
		return kResultUnknown;
	else if (allChecks.count() == 1) {
		ProbeResult probeResult = static_cast<ProbeResult>(std::countr_zero(allChecks.to_ulong()));
		ResMan.setPathPrefix(_pathPrefixes[probeResult]); // Set the final path prefix
		return probeResult;
	}
	else
		throw CreateException("Ambiguous archives: probe failed to recognise a certain game");
}
```
I previously opted in for power-of-two values specifically because regular values would be painful to convert from bitset mask back when project was using C++17 standard - would have to write the corresponding function myself, which would be a bit too much for the probing mechanism PR (I thought).

Reply:
Just my two cent, you could put the enum value into a set and would not need a bitset

Reply:
Thanks @maaxxaam @Nostritius for your suggestions. 
In the last commits, the ProbeResult enum has been updated to use sequential numbers so that it can be used as an index into the array. 
The enum order has been changed so that the first game starts from index 0 in the the array. 
I also updated the code to use set rather than bitset.
The return type of the checkXYZ functions have been changed from ProbeResult to bool to simplify a little bit the updated performProbe - this also better fits the purpose of the check functions.
While I initialy used a constexpr for the _pathPrefixes, I had to change it back to const because the mac compiler specified in the github workflow still has difficulty with std::string as constexpr. It worked fine with the other compilers. As a side-note, I recently had to update my gcc v11 to a newer version to be able to compile the latest master of OpenAWE due to some other commits that relied on constexpr std::string.